### PR TITLE
[CU-2hr9vn9] Andres/staking proptests

### PR DESCRIPTION
**Pull request by andresmechali**
## Issue
https://app.clickup.com/t/2hr9vn9

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2hr9vn9 Staking-rewards pallet: Proptest for staking pallet [day 1 to do]](https://app.clickup.com/t/2hr9vn9)

File: code/parachain/frame/staking-rewards/src/test/mod.rs [hunk 1]
```diff
@@ -1326,6 +1333,83 @@ fn duration_presets_are_required() {
...
+	use super::*;
+	use crate::Error;
+	use composable_tests_helpers::prop_assert_noop;
+
+	proptest! {
+		#![proptest_config(ProptestConfig::with_cases(10000))]
+
+		#[test]
+		fn stake_should_work(
+			amount in MINIMUM_STAKING_AMOUNT..u128::max_value(),
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/test/mod.rs:1346]:
```suggestion
			amount in MINIMUM_STAKING_AMOUNT..u128::MAX,
```

File: code/parachain/frame/staking-rewards/src/test/mod.rs [hunk 2]
```diff
@@ -1326,6 +1333,83 @@ fn duration_presets_are_required() {
...
+					duration_preset,
+				));
+
+				Ok(())
+			})?;
+		}
+
+		#[test]
+		fn stake_should_not_work_with_low_amounts(
+			amount in 1_u128..MINIMUM_STAKING_AMOUNT-1
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/test/mod.rs:1379]:
```suggestion
			amount in 0_u128..MINIMUM_STAKING_AMOUNT-1
```

File: code/parachain/frame/staking-rewards/src/test/mod.rs [hunk 3]
```diff
@@ -1523,6 +1529,247 @@ fn duration_presets_are_required() {
...
+
+mod extend_proptests {
+	use super::*;
+
+	proptest! {
+		#![proptest_config(ProptestConfig::with_cases(10000))]
+
+		#[test]
+		fn extend_should_work(
+			amount in 0_u128..PICA::units(200),
```
Comment (benluelo) [inline code/parachain/frame/staking-rewards/src/test/mod.rs:1739]: This could probably be a significantly larger value than 200 pica

Comment (andresmechali) [reply code/parachain/frame/staking-rewards/src/test/mod.rs:1739]: Yes, you are right. I set that value as I mint those PICA before extending. Any idea about what a reasonable value might be?

Comment (benluelo) [reply code/parachain/frame/staking-rewards/src/test/mod.rs:1739]: maybe something like 1 million? just an arbitrary high number

**Review by vimukthi-git — APPROVED** [review]
LGTM

File: code/parachain/frame/staking-rewards/proptest-regressions/test/mod.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Seeds for failure cases proptest has generated in the past. It is
+# automatically read and these particular cases re-run before any
+# novel cases are generated.
+#
+# It is recommended to check this file in to source control so that
+# everyone who runs the test benefits from these saved cases.
+cc e9c54402dec866a96921319f195004cdce8fe2d93911459a42103768990eeea6 # shrinks to network_id = 1, amm_id = 0, start_block = 1
+cc f60478cbb9380be7fea6c421e8ce2ab80edabce698b465c9669145deb780b19f # shrinks to amount = 170141183460469231731687303715884105228
+cc da2d5615192fd7056f17bb99e4904d6a0d385252d7a2fb795e9071a55130be39 # shrinks to amount = 336913234575186597488489710328483377680
+cc 3ea68462befd66841315d4b585ba1f548f7cdde9e6184766818c0ca93b874b5e # shrinks to parts = 990000
```
Comment (PoisonPhang) [inline code/parachain/frame/staking-rewards/proptest-regressions/test/mod.txt:10]: Did we intend to commit this file?

Comment (andresmechali) [reply code/parachain/frame/staking-rewards/proptest-regressions/test/mod.txt:10]: I would guess that we don't, but there are a few files like this already commited. I will remove it anyway.

**Comment by andresmechali** [comment]
@mergify refresh

**Pull request merged**